kubeflow/core/jupyterhub.libsonnet

```diff
@@ -16,6 +14,16 @@
   ],
 
   parts(namespace):: {
+    jupyterHubConfigMap(jupyterHubAuthenticator, disks): {
+      local util = import "kubeflow/core/util.libsonnet",
+      local diskNames = util.toArray(disks),
+      local kubeSpawner = $.parts(namespace).kubeSpawner(jupyterHubAuthenticator, diskNames),
+      local cm = if std.length(diskNames) == 0 then
+        $.parts(namespace).jupyterHubConfigMapWithoutVolumes
```

Comment:
jupyterHubConfigMapWithoutVolumes takes a parameter named spawner, but here you are not passing any parameter to jupyterHubConfigMapWithoutVolumes

Reply:
both parts of the if logic return a function which is then called with spawner on line # 24. 
Original logic is similar.

Reply:
The first part of the if logic returns a function which takes `spawner` as a parameter, but the second part `$.parts(namespace).jupyterHubConfigMapWithVolumes(diskNames)` returns an object, not a function.